supervisor/api/backups.py

```diff
@@ -473,6 +473,11 @@ async def download(self, request: web.Request):
 
         _LOGGER.info("Downloading backup %s", backup.slug)
         filename = backup.all_locations[location][ATTR_PATH]
+        # If the file is missing, return 404 and trigger reload of location
+        if not filename.is_file():
```

Comment:
Technically, this is file I/O in the asyncio event loop. But for "small" I/O like file existence check we do that in many other places, so I don't think we need to depart from that pattern today. Just reminded me again: We should try to move all access into executor threads. Maybe we should consider using aiofile :thinking: